quiz-app-api/app/controllers/api/UserController.js

```diff
@@ -1,107 +1,130 @@
 /*jshint esversion: 6 */
-const ResourceController = require("../ResourceController");
-const {
-  User 
-} = require("../../models");
-var bcrypt = require("bcryptjs");
-var jwt = require("jwt-simple");
-var config = require("../../../config/config.js");
+const ResourceController = require('../ResourceController');
+const { User } = require("../../models");
+const bcrypt = require("bcryptjs");
+const jwt = require('jsonwebtoken')
+const config = require('../../../config/config.js');
 
 
 class UserController extends ResourceController {
   constructor(...args) {
     super(...args);
   }
-}
 
-var uc = new UserController(User);
-user = {
-  signup: (req, res, next) => {
-    console.log(req.body);
-    let user = new User({
-      "name": req.body.name,
-      "email": req.body.email,
-      "cgiCode": req.body.cgiCode,
-      "photo": req.body.photo,
-      "userRole": req.body.userRole
-    });
-    bcrypt.hash(req.body.password, 10, (err, hash) => {
-      user.password = hash;
-      user.save((err, user) => {
-        if (err) {
-          console.error(err);
+  signup(req, res) {
+    // If the email already exists, don't signup a new email
+    User.findOne({ email: req.body.data.email })
+      .exec()
+      .then(result => {
+        if (result) {
+          return res.status(409).json({
+            message: 'Email Already Exists',

```

Comment:
Good for now, but later let's create a method which returns appropriate "message text" based on status code. So if someone wants to change message only one place change is required